File: Data Structures & Algorithms/trapping-rain-water/submission-7.java
class Solution {
    public int trap(int[] height) {
        int res = 0;
        
        for (int i = 0; i < height.length; i++) {
            int leftMax = 0, rightMax = 0;

            for (int l = 0; l < i; l++) {
                leftMax = Math.max(leftMax, height[l]);
            }

            for (int r = i + 1; r < height.length; r++) {
                rightMax = Math.max(rightMax, height[r]);
            }

            int currAmt = Math.min(leftMax, rightMax) - height[i];
            if (currAmt > 0) {
                res += currAmt;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-7.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        res = nums[0]
        for i in range(len(nums)):
            product = 1
            for j in range(i, len(nums)):
                product *= nums[j]
                res = max(res, product)

        return res


        